local/rewrite_configuration.py

```diff
@@ -0,0 +1,124 @@
+import os
+import pathlib
+import sys
+
+import yaml
+
+UNSUPPORTED_FEATURES = ["write_feast_features"]
+
+_NEW_CONFIG_NAME = "config.yaml.tmp"
+_DATA_DIR = pathlib.Path("/data")
+_OUTPUT_DIR = pathlib.Path("/output")
+
+config_file = sys.argv[1]
+
+with open(config_file, "rt") as f:
+    full_config = yaml.load(f, yaml.SafeLoader)
+
+
+def validate_config(sub_config):
+    try:
+        for k, v in sub_config.items():
+            if k in UNSUPPORTED_FEATURES:
+                raise ValueError(f"{k} is not supported in Docker execution mode.")
+            if isinstance(v, dict):
+                validate_config(v)
+    except AttributeError:
+        print(sub_config)
+
+
+validate_config(full_config)
+
+input_data_paths = []
+
+
+def add_path(path: str):
+    if path.startswith("dbfs:") or path.startswith("s3:"):
+        raise ValueError(f"Only local paths are supported: {path}")
+    input_data_paths.append(path)
+
+
+def find_input_data_paths(sub_config):
+    for k, v in sub_config.items():
+        if k == "read_dataset":
+            file_path = v["file_path"]
+            add_path(file_path)
```

Comment:
Along with this, is there any concerns related to other data ingest functions, i.e `join_dataset`, `concatenate_dataset`? As we know, they are all in the ETL data workflow, maybe we need to rule those functions out as well?

Reply:
There is nothing special about these functions. All of them contain a key `read_dataset` that contains a key `file_path` which we need to adapt. (The demo workflow we are running contains both a `join_dataset` and a `concatenate_dataset` block, both of which are processed fine.)

For more information, see the documentation: https://docs.anovos.ai/using-anovos/config_file.html

Reply:
Thanks for the explanation @ionicsolutions . This is resolved and closed for now.